core/src/com/threecubed/auber/World.java

```diff
@@ -436,4 +443,35 @@ public void checkForEndState() {
       game.setScreen(new GameOverScreen(game, true));
     }
   }
+
+  public enum Abilities {
+    superspeed,
+    invisibility,
+    invincibility,
+    insta_beam,
+    vision;
+
+    private static final List<Abilities> ABILITIES = Collections.unmodifiableList(Arrays.asList(values()));
+    private static final int length = ABILITIES.size();
+    private static final Random rng = new Random();
+
+    public static Abilities randomAbility() {
+      return ABILITIES.get(rng.nextInt(length));
+    }
+  }
+
+  /**
+   * Spawns in a random buff in the position of a random entity (Civilians only)
+   * */
+
+  public void spawnBuff() {
+    GameEntity ent = this.getEntities().get(this.randomNumberGenerator.nextInt(this.getEntities().size()));
+    while (!(ent instanceof Civilian)) {
+      ent = this.getEntities().get(this.randomNumberGenerator.nextInt(this.getEntities().size()));
+
+    }
```

Comment:
Could be rewritten as a do-while
```suggestion
    GameEntity ent;
    do {
      ent = this.getEntities().get(this.randomNumberGenerator.nextInt(this.getEntities().size()));
    } while (!(ent instanceof Civilian))
```

Reply:
Looking at this now, it'll hardlock the game if there are no civilians - but there are no cases like that... right?

Reply:
I'll commit it for now, and add a break clause that will trigger if the number of entities in game is less that the player add the infiltrators. Then, as long as civilians don't despawn it should be fine

Reply:
[Potentially fixed in this commit](https://github.com/2020-eng1-team2/auber-pt2/commit/8874d9939413d4869ada1256ef35d0e7f6216443)